osu.Game.Rulesets.Mania/Difficulty/Skills/Strain.cs

```diff
@@ -4,98 +4,281 @@
 #nullable disable
 
 using System;
-using osu.Framework.Utils;
 using osu.Game.Rulesets.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Difficulty.Skills;
 using osu.Game.Rulesets.Mania.Difficulty.Preprocessing;
 using osu.Game.Rulesets.Mods;
 
+/* Rules of Strain Calculation
+ * 1) The current strain value can only depend on past notes
+ * 2) Notes on the same offset may see different strain values, as column order sorting is not guaranteed.
+ * 3) 1st hit object is OMITTED as it acts as a reference for the following note.
+ *
+ * E.g. A:(100ms @ Col 1), B:(200ms @ Col 1), C:(200ms @ Col 2)
+ * The following sequences are possible: B C, C B.
```

Comment:
Sequences of what? Strain accumulation?

Reply:
It's meant to show that the notes fed into `Strain.cs` is not deterministic if notes are on the same time, I'll reword